src/sentry/tasks/integrations/migrate_alert_rules.py

```diff
@@ -0,0 +1,109 @@
+from django.db import router, transaction
+
+from sentry.models import Integration, Project, Rule
+from sentry.models.integrations.organization_integration import OrganizationIntegration
+from sentry.plugins.base import plugins
+from sentry.services.hybrid_cloud.integration.service import integration_service
+from sentry.tasks.base import instrumented_task, retry
+from sentry.tasks.integrations import logger
+
+ALERT_LEGACY_INTEGRATIONS = {
+    "id": "sentry.rules.actions.notify_event.NotifyEventAction",
+    "name": "Send a notification (for all legacy integrations)",
+}
+
+
+@instrumented_task(
+    name="sentry.tasks.integrations.migrate_alert_rules",
+    queue="integrations",
+    default_retry_delay=60 * 5,
+    max_retries=5,
+)
+@retry(exclude=[Integration.DoesNotExist])
+def migrate_alert_rules(integration_id: int, organization_id: int) -> None:
+    from sentry_plugins.opsgenie.plugin import OpsGeniePlugin
+
+    integration, organization_integration = integration_service.get_organization_context(
+        organization_id=organization_id, integration_id=integration_id
+    )
+    if not integration:
+        raise Integration.DoesNotExist
+    if not organization_integration:
+        raise OrganizationIntegration.DoesNotExist
+
+    seen_keys = {team["integration_key"] for team in organization_integration.config["team_table"]}
+
+    for project in Project.objects.filter(organization_id=organization_id):
```

Comment:
@mifu67 @leeandher This will need to be refactored later on to support the split databases for hybrid cloud, as the projects and alert rules tables will not co-exist with the integration and organization integration tables.

Reply:
followed up with a work around in my review!